clang/test/CheckedC/inferred-bounds/widened-bounds-check.c

```diff
@@ -13,7 +13,7 @@ void f1() {
   {}
 
   if (*p) {
-    p++; // expected-error {{declared bounds for p are invalid after assignment}}
+    p++; // expected-error {{declared bounds for 'p' are invalid after statement}}
```

Comment:
At the statement `p++`, `p` no longer has widened bounds (the modification of `p` kills the widened bounds of `p`).

`p` has the original value `p - 1`. At the end of the statement `p++`, the updated observed bounds of `p` are `bounds(p - 1, p - 1)`. These bounds do not imply the declared bounds `(p, p)`.